okhttp/src/jvmMain/kotlin/okhttp3/internal/concurrent/TaskQueue.kt

```diff
@@ -180,19 +184,19 @@ class TaskQueue internal constructor(
    * be removed from the execution schedule.
    */
   fun cancelAll() {
-    this.assertThreadDoesntHoldLock()
+    lock.assertThreadDoesntHold()
 
-    synchronized(taskRunner) {
+    taskRunner.lock.withLock {
       if (cancelAllAndDecide()) {
         taskRunner.kickCoordinator(this)
       }
     }
   }
 
   fun shutdown() {
-    this.assertThreadDoesntHoldLock()
+    lock.assertThreadDoesntHold()
```

Comment:
Rename assertNotHeld ?

(Previously the fact that `Any` is a lock is weird, and so this function got a deservingly weird name. Now that it’s a function on Lock, it can have a reasonable name?)